### ISSUE TITLE
Fix for 'var-naming warns on variables with dynamically set names' removed

### ISSUE THREAD

**Issue by erichiller** [ansible/ansible-lint#4296]
The same issue as reported in #1680 (`var-naming` warns on variables with dynamically set names) and fixed in #1988 (See [file diff](https://github.com/ansible/ansible-lint/pull/1988/files#diff-87fedcd1d0c351475fbdbca0079a682d413b9d9254fb6f3ca07e2bd2a21d8da7)) was then later undone in #3422 , but that commit does not appear to have been about `var-naming[no-jinja]`. In fact the comment from the original fix is still present,
```python
        # We want to allow use of jinja2 templating for variable names
        if "{{" in ident:
```
so I can only assume that the removal of the `return False` was not intentional.

Can the behaviour post #1988 be restored?

**Comment by shatakshiiii** [comment]
@ssbarnea ^^
